src/ServerError.js

```diff
@@ -0,0 +1,26 @@
+import Parse from 'parse';
+
+/**
+ * Constructs a new ServerError object with the given message and source.
+ *
+ * The message is an error that is intended for the server logs
+ * while the source message is intended for the server logs to aid a developer
+ *
+ * @class ServerError
+ * @constructor
+ * @param {ErrorCode} code the Parse.Error code
+ * @param {String} message will be sent to clients
+ * @param {String} serverErrorLog additional information
+ *                  about the error for logging on the server
+ * @param {String} stack the stack trace
+ */
+export default class ServerError extends Parse.Error {
```

Comment:
well, first off: `myObj.constructor.name` as a hack to get the classname of an object is my new favorite javascriptism.

so, I can't explain exactly why it is happening, but I can tell you what I observe.  When I extend `ServerError` from `Error`, the `err` param of `middlewares.handleParseError` is an `Error`, not a `ServerError` as one would expect.  If I extend `ServerError` from `Parse.Error` then the `err` param is a `ServerError` as one would expect.

So an interesting thing to note is that the `Error` object does have all of the members of the `ServerError`, but it fails the test `err instanceof ParseServer.Error` which causes bad things to ensue.

So I could conceivably add a reliable marker to the `ParseServer` class that I could test, say add a member like `isServerError = true` or some such, but that doesn't appeal to me.

In any event, open to another approach and also glad to go with the marker approach if you prefer it.

PS could def use some help with the failing unit test if you have a clue....